src/machine/board_xiao-rp2040.go

```diff
@@ -0,0 +1,78 @@
+//go:build xiao_rp2040
+// +build xiao_rp2040
+
+// This file contains the pin mappings for the Seeed XIAO RP2040 boards.
+//
+// XIAO RP2040 is a microcontroller using the Raspberry Pi RP2040 chip.
+//
+// - https://wiki.seeedstudio.com/XIAO-RP2040/
+//
+package machine
+
+// Digital Pins
+const (
+	D0  Pin = GPIO26
+	D1  Pin = GPIO27
+	D2  Pin = GPIO28
+	D3  Pin = GPIO29
+	D4  Pin = GPIO6
+	D5  Pin = GPIO7
+	D6  Pin = GPIO0
+	D7  Pin = GPIO1
+	D8  Pin = GPIO2
+	D9  Pin = GPIO3
+	D10 Pin = GPIO4
+)
+
+// Analog pins
+const (
+	A0 Pin = D0
+	A1 Pin = D1
+	A2 Pin = D2
+	A3 Pin = D3
+)
+
+// Onboard LEDs
+const (
+	NEOPIXEL = GPIO12
+
+	LED_RED   = GPIO17
+	LED_GREEN = GPIO16
+	LED_BLUE  = GPIO25
+)
+
+// I2C pins
+const (
+	I2C0_SDA_PIN Pin = D4
+	I2C0_SCL_PIN Pin = D5
+
+	I2C1_SDA_PIN Pin = NoPin
+	I2C1_SCL_PIN Pin = NoPin
+)
+
+// SPI pins
+const (
+	SPI0_SCK_PIN Pin = D8
+	SPI0_SDO_PIN Pin = D9
+	SPI0_SDI_PIN Pin = D10
+
+	SPI1_SCK_PIN Pin = NoPin
+	SPI1_SDO_PIN Pin = NoPin
+	SPI1_SDI_PIN Pin = NoPin
+)
+
+// Onboard crystal oscillator frequency, in MHz.
+const (
+	xoscFreq = 12 // MHz
+)
+
+// USB CDC identifiers
+const (
+	usb_STRING_PRODUCT      = "XIAO RP2040"
+	usb_STRING_MANUFACTURER = "Seeed"
+)
+
+var (
+	usb_VID uint16 = 0x2e8a
+	usb_PID uint16 = 0x0003
```

Comment:
Looking at the following, the PID of the XIAO RP2040 appears to be 0x000A.
Perhaps it would be better to adapt it to the Arduino environment.
However, I am not sure if 0x000A is really the correct value.
If 0x0003 is the value used by the Arduino, then the PID in #2936 may also need to be changed slightly.

https://github.com/earlephilhower/arduino-pico/blob/master/boards.txt

Reply:
In bootloader mode lsusb output is : 

```
Bus 001 Device 024: ID 2e8a:0003 
```

This is the as badger2040 : https://github.com/tinygo-org/tinygo/blob/release/src/machine/board_badger2040.go#L80

Reply:
I confirmed this with xiao-samd21 and xiao-rp2040.

* xiao samd21
  * When built on Arduino, it is 2886/802F.
  * The bootloader is 2886/002F.
  * `board_xiao.go` is set to 2886/802F.
* xiao rp2040
  * When built on Arduino, it is 2E8A/000A.
  * The bootloader is 2886/0003.

I think 2E8A/000A is correct this time, since at least tinygo-flashed state is not bootloader.

https://github.com/tinygo-org/tinygo/blob/b65447c7d567eea495805656f45472cc3c483e03/targets/xiao.json#L5
https://github.com/tinygo-org/tinygo/blob/b65447c7d567eea495805656f45472cc3c483e03/src/machine/board_xiao.go#L96-L99